Modules/Office365DSC/DSCResources/MSFT_SCDLPComplianceRule/MSFT_SCDLPComplianceRule.psm1

```diff
@@ -0,0 +1,559 @@
+        Set-DLPComplianceRule @UpdateParams
+    }
+    elseif (('Absent' -eq $Ensure) -and ('Present' -eq $CurrentRule.Ensure))
+    {
+        Write-Verbose "Rule {$($CurrentRule.Name)} already exists but shouldn't. Deleting Rule."
+        Remove-DLPComplianceRule -Identity $CurrentRule.Name -Confirm:$false
+    }
+}
+
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Name,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Policy,
+
+        [Parameter()]
+        [ValidateSet("InOrganization","NotInOrganization","None")]
+        [System.String[]]
+        $AccessScope,
+
+        [Parameter()]
+        [System.Boolean]
+        $BlockAccess,
+
+        [Parameter()]
+        [ValidateSet("All", "PerUser")]
+        [System.String]
+        $BlockAccessScope,
+
+        [Parameter()]
+        [System.String]
+        $Comment,
+
+        [Parameter()]
+        [Microsoft.Management.Infrastructure.CimInstance]
+        $ContentContainsSensitiveInformation,
+
+        [Parameter()]
+        [System.String[]]
+        $ContentPropertyContainsWords,
+
+        [Parameter()]
+        [System.Boolean]
+        $Disabled,
+
+        [Parameter()]
+        [System.String[]]
+        $GenerateAlert,
+
+        [Parameter()]
+        [System.String[]]
+        $GenerateIncidentReport,
+
+        [Parameter()]
+        [ValidateSet("All", "Default", "Detections", "DocumentAuthor", "DocumentLastModifier", "MatchedItem", "RulesMatched", "Service", "Severity", "Title")]
+        [System.String[]]
+        $IncidentReportContent,
+
+        [Parameter()]
+        [ValidateSet("FalsePositive", "WithoutJustification", "WithJustification")]
+        [System.String[]]
+        $NotifyAllowOverride,
+
+        [Parameter()]
+        [System.String]
+        $NotifyEmailCustomText,
+
+        [Parameter()]
+        [System.String]
+        $NotifyPolicyTipCustomText,
+
+        [Parameter()]
+        [System.String[]]
+        $NotifyUser,
+
+        [Parameter()]
+        [ValidateSet("Low", "Medium", "High", "None")]
+        [System.String]
+        $ReportSeverityLevel,
+
+        [Parameter()]
+        [ValidateSet("Ignore", "RetryThenBlock")]
+        [System.String]
+        $RuleErrorAction,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Write-Verbose -Message "Testing configuration of DLPComplianceRule for $Name"
+
+    $CurrentValues = Get-TargetResource @PSBoundParameters
+
+    Write-Verbose -Message "Current Values: $(Convert-O365DscHashtableToString -Hashtable $CurrentValues)"
```

Comment:
Is this line really necessary? The last line in the Get method also outputs this data, resulting in duplicate output.